mri/reconstruct/fourier.py

```diff
@@ -208,3 +208,139 @@ def adj_op(self, x):
+            What FFT version to be used for 2D non uniform FFT
+        platform: string, 'cpu', 'multi-cpu' or 'gpu' default 'gpu'
+            string indicating which hardware platform will be used to
+            compute the NUFFT. works only if fft_type=='NUFFT'
+        """
+        self.num_slices = shape[2]
+        self.shape = shape
+        # Sort the incoming data based on Z, Y then X coordinates
+        # This is done for easier stacking
+        self.sort_pos = np.lexsort(tuple(samples[:, i]
+                                         for i in np.arange(3)))
+        samples = samples[self.sort_pos]
+        plane_samples, self.z_samples = self.get_stacks(samples)
+        if fft_type == 'NUFFT':
+            self.FT = NUFFT(samples=plane_samples, shape=shape[0:2],
+                            platform=platform)
+        elif fft_type == 'NFFT':
+            self.FT = NFFT(samples=plane_samples, shape=shape[0:2])
+
+    def get_stacks(self, samples):
+        """ Function that converts an incoming 3D kspace samples
+            and converts to stacks of 2D. This function also checks for
+            any issues incoming k-space pattern and if the stack property
+            is not satisfied.
+            Stack Property:
+                The k-space locations originate from a stack of 2D samples
+        Parameters
+        ----------
+        samples: np.ndarray
+            the mask samples in the 3D Fourier domain.
+
+        Returns
+        ----------
+        plane_samples: np.ndarray
+            A 2D array of samples which when stacked gives the 3D samples
+        z_samples: np.ndarray
+            A 1D array of z-sample locations
+        """
+        self.first_stack_len = np.size(np.where(samples[:, 2]
+                                                == np.min(samples[:, 2])))
+        self.acq_num_slices = int(len(samples) / self.first_stack_len)
+        stacked = np.reshape(samples, (self.acq_num_slices,
+                                       self.first_stack_len, 3))
+        z_expected_stacked = np.reshape(np.repeat(stacked[:, 0, 2],
+                                                  self.first_stack_len),
+                                        (self.acq_num_slices,
+                                         self.first_stack_len))
+        if np.mod(len(samples), self.first_stack_len) \
+                or not np.all(stacked[:, :, 0:2] == stacked[0, :, 0:2]) \
+                or not np.all(stacked[:, :, 2] == z_expected_stacked):
+            raise ValueError('The input must be a stack of 2D k-Space data')
+        plane_samples = stacked[0, :, 0:2]
+        z_samples = stacked[:, 0, 2]
+        z_samples = z_samples[:, np.newaxis]
+        return plane_samples, z_samples
+
+    def op(self, data):
+        """ This method calculates Fourier transform.
+        Parameters
+        ----------
+        img: np.ndarray
+            input image as array.
+
+        Returns
+        -------
+        result: np.ndarray
```

Comment:
I wasn't paying attention in the last PR probably, but expliciting a name for the return value isn't a good practice imho, especially when it doesn't match the name in code (happens in other places in this PR).

Reply:
Why isn't that a good practice?

Reply:
Hmm, I don't exactly follow your point or don't agree with it. 
Are you saying that in documentation we should not give the name of return variable? 

In my opinion, we need to give such names and also tell what the variable is. The name is very important especially when the function returns multiple variables

Reply:
It's not necessarily a bad practice, but you just don't need it, you need the names of the arguments because they can be used to call the function, you don't need to know the name of the return variable which is internal to the function. In short it's not needed, so we shouldn't have (but I am not strongly against it).

When the function returns multiple variables, it might be better for clarity's sake. Even still, the name should match that of the returned variable.